src/librados/RadosClient.cc

```diff
@@ -56,7 +56,9 @@ namespace ca = ceph::async;
 namespace cb = ceph::buffer;
 
 librados::RadosClient::RadosClient(CephContext *cct_)
-  : Dispatcher(cct_->get()) {
+  : Dispatcher(cct_->get()),
+    cct_deleter{cct, [](CephContext *p) {p->put();}}
```

Comment:
Thinking about this again: should we not refactor Dispatcher to get/put its own ref to cct so this isn't necessary?

Reply:
> Thinking about this again: should we not refactor Dispatcher to get/put its own ref to cct so this isn't necessary?

the same idea struck me as well when creating this PR. since we could have 1-to-n and n-to-1 mapping between Dispatcher and messenger. and the ownership is like a double-edge sward, because it may lead us to leaks caused by cyclic reference. so i'd defer that to another more adventurous PR.

> 
> Edit: I think there should be two refs here: one for Dispatcher and one for RadosClient. 

because `RadosClient` is a `Dispatcher`. it'd be weird we have two references pointing to the same object in parent class *and* in child class.

> Also, it's odd we don't use intrusive_ptr but I guess there's some technical reason for that.

yeah, in the more adventurous PR, i will try to use `intrusive_ptr`. otherwise we will have to have two references..